### PR TITLE
ci: names for jobs

### PR DESCRIPTION
**Pull request by janpio**

### CODE REVIEW

File: .github/workflows/test-template.yml [hunk 1]
```diff
@@ -947,6 +961,7 @@ jobs:
   # minimize the time spent waiting for a free runner.
   #
   no-docker-client-functional:
+    name: TODO
```
Comment (janpio) [inline .github/workflows/test-template.yml:964]: TODO Needs a name

File: .github/workflows/test-template.yml [hunk 2]
```diff
@@ -1017,6 +1032,7 @@ jobs:
           secret: ${{ secrets.BUILDPULSE_SECRET_ACCESS_KEY }}
 
   no-docker:
+    name: TODO
```
Comment (janpio) [inline .github/workflows/test-template.yml:1035]: TODO Needs a name

File: .github/workflows/test-template.yml [hunk 3]
```diff
@@ -186,7 +188,8 @@ jobs:
   #
   # CLIENT (functional tests with mini-proxy)
   #
-  client-dataproxy:
+  client-miniproxy:
+    name: Client (functional tests with mini-proxy)
```
Comment (Jolg42) [inline .github/workflows/test-template.yml:192]: nice!

File: .github/workflows/test-template.yml (hunk 1 above)
Comment (Jolg42) [reply .github/workflows/test-template.yml:964]:
```suggestion
    name: Client (functional tests running on Windows/macOS, no-docker)
```
Maybe?

File: .github/workflows/test-template.yml (hunk 2 above)
Comment (Jolg42) [reply .github/workflows/test-template.yml:1035]:
```suggestion
    name: All packages (running on Windows/macOS, no-docker)
```

**Review by Jolg42 — APPROVED** [review]
✨

**Review by jkomyno — APPROVED** [review]

**Review by millsp — APPROVED** [review]

**Pull request merged**